### ISSUE TITLE
Remove all references of mktap in the docs

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3393]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#3393|
|Type|task|
|Created|2008-08-19 17:03:05Z|
|Branch|https://github.com/twisted/twisted/tree/remove-mktap-3393|

References recommending the use of mktap in the documentation have to go, it was deprecated [ticket:2013 2 years] ago.

Attachments:

* [nomktap.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/8ff/8ff080f8aba1bd4557bf641b3684c8e5ec346bff/b3328130be6063b86491d57457154a9a275f6d7a.diff) (5977 bytes) - added by kehander1 on 2008-10-14 17:59:53Z - Removal of Removal of mktap from using-twistedweb and philosophy (v2)
* [design-mktap.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/8ff/8ff080f8aba1bd4557bf641b3684c8e5ec346bff/caaab03afec53dfcbcc58862c59cdf8ea0fdee90.patch) (787 bytes) - added by kehander1 on 2008-10-21 21:42:02Z - Removal of mktap from design.xhtml
* [nomktap-plus.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/8ff/8ff080f8aba1bd4557bf641b3684c8e5ec346bff/2c5c142b7029a41f47d926a16c6dcf1520636e7e.diff) (11299 bytes) - added by kehander1 on 2008-10-28 17:16:20Z - Removal of mktap from more docs, plugins, and man pages

<details><summary>Searchable metadata</summary>

```
trac-id__3393 3393
type__task task
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__branches_remove_mktap_3393 branches/remove-mktap-3393
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1219165385000000 1219165385000000
changetime__1225224745000000 1225224745000000
version__None None
owner__ 
cc__thijs cc__kehander1 cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_4"></a>|@ralphm set owner to kehander1|
|-|-|

These changes mostly look good to me. Some nits:

 * In `writing-standard.xhtml` you change one link to include the `format=raw` parameter, while leaving the other untouched. I don't think these links should be changed at all. The only reason to add them is because of the way they are currently made available through Trac's code browser. I believe we should just export all the documentation files regularly (on each commit?). The project documentation is still in the historic `/projects` tree, but the policy docs aren't. This might relate to [#3438](https://github.com/twisted/twisted/issues/3438).
 * With the removal of `mktap` in `using-twistedweb.xhtml`, I think it might be good to introduce `twistd web` somewhere at the top of the document. That makes the document more readable.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

> I don't think these links should be changed at all. The only reason to add them is because of the way they are currently made available through Trac's code browser.

Yep, you're right, they shouldn't be changed.  Doing so will break the non-trac published documentation.

**Comment by twisted-trac** [comment]
|<img alt="kehander1's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|kehander1 commented|
|-|-|

Oops, shouldn't have done that.  Fixed.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_8"></a>|@ralphm commented|
|-|-|

Looking good now. This can be merged.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_9"></a>|@ralphm set status to `closed`|
|-|-|

(In [24977]) Remove several references to mktap from the documentation.

Author: kehander1
Reviewer: ralphm
Fixed [#3393](https://github.com/twisted/twisted/issues/3393).

**Comment by twisted-trac** [comment]
|<img alt="kehander1's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_10"></a>|kehander1 set status to `reopened`|
|-|-|

Bother, missed one.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun commented|
|-|-|

The only way to re-open tickets is to revert the change which closed them.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun commented|
|-|-|

(In [25002]) Revert r24977 - the fix was incomplete

refs [#3393](https://github.com/twisted/twisted/issues/3393)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun commented|
|-|-|

(In [25003]) Branching to 'remove-mktap-3393'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_16"></a>|@exarkun commented|
|-|-|

(In [25009]) apply design-mktap.patch

refs [#3393](https://github.com/twisted/twisted/issues/3393)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_17"></a>|@exarkun set owner to kehander1<br>@exarkun set status to `new`|
|-|-|

The `--help` output for many `twistd` subcommands still refer to mktap.  They do so in a way which has always been bogus, but most people seem either to notice or seem not to think worth comment:

```
exarkun@charm:~$ twistd conch --help
Usage: twistd [options] conch Usage: mktap conch [-i <interface>] [-p <port>] [-d <dir>]
Options:
...
```

This is nonsensical and would more appropriately be something along the lines of:

```
exarkun@charm:~$ twistd conch --help
Usage: twistd [options] conch [-i <interface>] [-p <port>] [-d <dir>]
Options:
...
```

This is controlled by the `synopsis` attribute of the `Options` subclass specified by the plugin.

Though the web2 docs also still refer to mktap, I'm not really concerned about them.  Not only is `mktap web2` deprecated, but we'll get rid of `twistd web2` as well as soon as we can.

However, doc/web/howto/web-development.xhtml also still refers to mktap, as do the man pages for `tap2deb`, `tap2rpm`, `tapconvert`, and `twistd`.

**Comment by twisted-trac** [comment]
|<img alt="kehander1's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_18"></a>|kehander1 removed owner|
|-|-|

Not sure what to do about tap2deb and tap2rpm, but this fixes a bunch of the other ones.  And of course there's still [#3246](https://github.com/twisted/twisted/issues/3246) .

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_19"></a>|@exarkun commented|
|-|-|

(In [25158]) Apply nomktap-plus.diff

refs [#3393](https://github.com/twisted/twisted/issues/3393)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_20"></a>|@exarkun commented|
|-|-|

(In [25159]) whitespace cleanup

refs [#3393](https://github.com/twisted/twisted/issues/3393)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_21"></a>|@exarkun commented|
|-|-|

(In [25160]) whitespace cleanup, remove a mostly duplicate section, make some other minor edits

refs [#3393](https://github.com/twisted/twisted/issues/3393)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_22"></a>|@exarkun commented|
|-|-|

(In [25161]) whitespace cleanup

refs [#3393](https://github.com/twisted/twisted/issues/3393)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_23"></a>|@exarkun commented|
|-|-|

(In [25162]) whitespace cleanup

refs [#3393](https://github.com/twisted/twisted/issues/3393)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_24"></a>|@exarkun commented|
|-|-|

(In [25163]) remove references to mktap from names howto

refs [#3393](https://github.com/twisted/twisted/issues/3393)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_25"></a>|@exarkun commented|
|-|-|

(In [25164]) mention that twisted application pickles are deprecated

refs [#3393](https://github.com/twisted/twisted/issues/3393)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_26"></a>|@exarkun commented|
|-|-|

(In [25165]) remove some more references to mktap

refs [#3393](https://github.com/twisted/twisted/issues/3393)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_27"></a>|@exarkun commented|
|-|-|

(In [25166]) la la la twistd is so great

refs [#3393](https://github.com/twisted/twisted/issues/3393)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_28"></a>|@exarkun commented|
|-|-|

(In [25167]) la la la twistd is so great

refs [#3393](https://github.com/twisted/twisted/issues/3393)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_29"></a>|@exarkun set owner to @exarkun|
|-|-|

Fixed some more issues.  Itamar reviewed the changes I made.  Going to merge the branch now.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_30"></a>|@exarkun set status to `closed`|
|-|-|

(In [25168]) Merge remove-mktap-3393

Author: kehander1, exarkun
Reviewer: ralphm, itamar
Fixes: [#3393](https://github.com/twisted/twisted/issues/3393)

Remove almost all references to mktap(1) from the Twisted documentation.

The finger tutorial still refers to mktap since it also contains an
old-style plugin which will not work with twistd.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
